test/grouping.jl

```diff
@@ -636,163 +636,95 @@ end
 
     # Only test that different by syntaxes work,
     # and rely on tests below for deeper checks
-    @test by(:c => sum, df, :a) ==
-        by(df, :a, :c => sum) ==
-        by(df, :a, (:c => sum,)) ==
+    @test by(df, :a, :c => sum) ==
+        by(:c => sum, df, :a) ==
+        by(df, :a, :c => sum => :c_sum) ==
+        by(:c => sum => :c_sum, df, :a) ==
         by(df, :a, [:c => sum]) ==
-        by(df, :a, c_sum = :c => sum) ==
+        by(df, :a, [:c => sum => :c_sum]) ==
         by(d -> (c_sum=sum(d.c),), df, :a) ==
         by(df, :a, d -> (c_sum=sum(d.c),))
 
-    @test by(:c => vexp, df, :a) ==
-        by(df, :a, :c => vexp) ==
-        by(df, :a, (:c => vexp,)) ==
+    @test by(df, :a, :c => vexp) ==
+        by(:c => vexp, df, :a) ==
+        by(df, :a, :c => vexp => :c_function) ==
+        by(:c => vexp => :c_function, df, :a) ==
         by(df, :a, [:c => vexp]) ==
-        by(df, :a, c_function = :c => vexp) ==
+        by(df, :a, [:c => vexp => :c_function]) ==
         by(d -> (c_function=vexp(d.c),), df, :a) ==
         by(df, :a, d -> (c_function=vexp(d.c),))
```

Comment:
Not sure why it wasn't covered here, but how about adding `by(df, :a, :c => c -> (c_function=vexp(c),))` and `by(:c => c -> (c_function=vexp(c),), df, :a)`? Same below in several blocks.

Reply:
I can add these tests.